src/effect.cpp

```diff
@@ -1160,10 +1159,15 @@ void load_effect_type( JsonObject &jo )
     if( jo.has_member( "name" ) ) {
         JsonArray jsarr = jo.get_array( "name" );
         while( jsarr.has_more() ) {
-            new_etype.name.push_back( jsarr.next_string() );
+            translation name;
+            if( !jsarr.read_next( name ) ) {
+                debugmsg( "Error reading effect names" );
```

Comment:
It would be better to call `jsarr.throw_error` to generate an exception along with a proper JSON context.